sdk/storage/Azure.Storage.DataMovement/src/TransferJobInternal.cs

```diff
@@ -332,43 +313,46 @@ public async virtual Task InvokeFailedArgAsync(Exception ex)
         /// </summary>
         public async Task JobPartEvent(TransferStatusEventArgs args)
         {
-            DataTransferStatus jobPartStatus = args.StorageTransferStatus;
+            DataTransferStatus jobPartStatus = args.TransferStatus;
             DataTransferStatus jobStatus = _dataTransfer._state.GetTransferStatus();
```

Comment:
Maybe we should save the state here locally instead of doing a bunch of `JobStatus.State` below. I also chose to save this locally in case it changes will this function is executing so best to save the state in case it changes.

This does present an interesting thing that sort of brings up some questions around naming. My suggestion would mean this line would be: `_dataTransfer._state.GetTransferStatus().State`. The two states here feels weird.

Reply:
Yeah the two states do seem weird. But I think I might have to remove the whole `DataTransferStateInternal` in a different PR, just to make sure I don't break anything doing that.

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/38537